src-tauri/src/lib.rs

```diff
@@ -327,6 +335,38 @@ async fn save_txt_files(
     Ok(())
 }
 
+#[tauri::command]
+async fn resolve_approval_request(
+    args: ResolveApprovalArgs,
+    state: tauri::State<'_, RwLock<State>>,
+) -> Result<(), String> {
+    println!("Resolving approval request");
+    let lock = state.read().await;
+
+    lock.handle
+        .as_ref()
+        .unwrap()
+        .resolve_approval(args.request_id.parse().unwrap(), args.accept)
+        .await
+        .to_string_result()?;
+    println!("Resolved approval request");
+
+    Ok(())
+}
+
+#[tauri::command]
+async fn initialize_handle(
```

Comment:
I don't think this needs be its own tauri command. There's no need for the frontend to be involved here at all?

Reply:
We should call `lock.set_handle(..)` in the `setup(...)` function. We can also remove it from `rpc.ts` then.